File: packages/leet_code/js_code/running-sum-of-1d-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    let acc = 0;
    return nums.map((value) => {
        acc = acc + value;
        return acc
    })
};


runningSum([1,2,3,4]) /* ? */ 

File: packages/leet_code/rust_code/src/problems/running_sum_of_1d_array.rs
pub fn running_sum_of_1d_array(nums: Vec<i32>) -> Vec<i32> {
    let mut acc = 0;
    nums.into_iter().map(|num| {
        acc = acc + num;
        acc
    }).collect()
}

#[test]
fn test() {
    assert_eq!(running_sum_of_1d_array(vec![1, 2, 3, 4]), vec![1, 3, 6, 10]);
    assert_eq!(running_sum_of_1d_array(vec![1,1,1,1,1]), vec![1, 2, 3, 4, 5]);
}
